Sprint4/step 1

i0ji opened the pull request:
almost all components are typed;
refactor of closing and opening order detail modal  logic;

Commit:
07.03.24/b:sprint3/step-1/v:0.1.9.3
global: added jest, node @types;
fix: typing components and API

Commit:
07.03.24/b:sprint3/step-1/v:0.1.9.3.1
global: multiply typings component replacement, stat to fix type issues
feature: add custom types file
feature: added declaration folder for interfaces. types and vars
fix: replace interfaces and types
fix: replace all request to api.ts
refactor: some api logic refactor, some request refactor

Commit:
08.03.24/b:sprint4/step-1/v:0.1.9.3.1
fix: small animation logic edit on profile page;
feature: typing

Commit:
11.03.24/b:sprint4/step-1/v:0.1.9.3.1

i0ji committed:
refactor: prevent order logic

i0ji committed:
fix: type issues in DND components and currentSlice
fix: types file improvements and optimization;

i0ji committed:
fix: type issues in DND components and currentSlice
fix: types file improvements and optimization;

i0ji committed:
typing

Commit:
12.03.24/b:sprint4/step-1/v:0.1.9.2
fix: multiply type issues edits

Commit:
fix: api.ts and App.tsx are typed

Commit:
modal typing

i0ji committed:
13.03.24/b:sprint4/step-1/v:0.1.9.1.3.2;
refactor: replace order logic from useModal hook to constructor;
fix: typing;
style: README small touch;

i0ji committed:
fix: typing
feature: added close logic to Warning modal

i0ji committed:
fix: typing

Commit:
fix:remove unused components; get Modal-wrapper into OrderDetails component and set  correct closing logic;

Commit:
fix: register type issue; CurrentIngredient type issue

gennady-bars commented on src/utils/check-response.ts:
```diff
@@ -1,6 +1,3 @@
-        throw new Error('Кое-чего случилось...');
-    }
-    return response.json();
+export const checkResponse = <T>(res: Response): Promise<T> => {
```
Отлично, что знаете про тип `Response` для ответа от сервера

gennady-bars commented on src/pages/LoginPage.tsx:
```diff
@@ -27,20 +26,26 @@ function LoginPage() {
+    // --------------- ERROR MESSAGE ---------------
+    const loginError = useSelector((state: RootState) => state.authSlice.loginError);
+    console.log(loginError)
     const handleLogin = (event: React.FormEvent<HTMLFormElement>) => {
```
Хорошо типизировали функции

gennady-bars commented on src/components/common/Modal/Modal.tsx:
```diff
@@ -19,17 +20,13 @@ export default function Modal({onClose, children}: {
 
     useEffect(() => {
-        setTimeout(() => {
             const closeOnEscapeKey = (e: KeyboardEvent) => {
```
Отлично, что знаете про тип `KeyboardEvent` для событий клавиатуры

gennady-bars commented on src/pages/LoginPage.tsx:
```diff
@@ -3,22 +3,21 @@ import styles from "pages/Pages.module.scss"
-import React, {Fragment, useEffect, useState} from "react";
+import React, {useState} from "react";
 import {Link} from "react-router-dom";
 import {useDispatch, useSelector} from "react-redux";
```
`import { useDispatch, useSelector } from 'react-redux';` такого больше не должно быть в проекте.  Хуки `useDispatch, useSelector` нужно типизировать **1 раз** в отдельном файле и заменить их во всем проекте на свои, а не из библиотеки `react-redux` .  Вот ссылка на документацию по их типизации https://redux.js.org/usage/usage-with-typescript#define-typed-hooks

gennady-bars commented on src/pages/LoginPage.tsx:
```diff
@@ -27,20 +26,26 @@ function LoginPage() {
     };
 
+    // --------------- ERROR MESSAGE ---------------
+    const loginError = useSelector((state: RootState) => state.authSlice.loginError);
```
`(state: RootState)` уже не нужно будет типизировать вручную стейт, когда замените везде хук `useSelector`. Нужно будет удалить дублирование типизации

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.



* Отлично, что нет типов `any`
* Отлично, что знаете про тип `Response` для ответа от сервера
* Отлично, что знаете про тип `KeyboardEvent` для событий клавиатуры
* Хорошо типизировали функции
* Хорошо типизировали компоненты


Удачного дальнейшего обучения.



* Я собрал заказ, а кнопка неактивная. Скрин https://disk.yandex.ru/i/U07b9xSKeQKBIQ . Пользователь не поймет, почему так. Неавторизованный пользователь **не должен** отправлять заказ на сервер. При клике на кнопку его нужно переводить на логин и после удачного логина обратно к заказу. 
* `import { useDispatch, useSelector } from 'react-redux';` такого больше не должно быть в проекте.  Хуки `useDispatch, useSelector` нужно типизировать **1 раз** в отдельном файле и заменить их во всем проекте на свои, а не из библиотеки `react-redux` .  Вот ссылка на документацию по их типизации https://redux.js.org/usage/usage-with-typescript#define-typed-hooks
* `(state: RootState)` уже не нужно будет типизировать вручную стейт, когда замените везде хук `useSelector`. Нужно будет удалить дублирование типизации

i0ji committed:
fix: declaration routes; type st; get rid of unused interfaces

i0ji committed:
14.03.23/b:sprint4/step-1/v:0.1.9.2
style: animation moved to separate component;
fix: declaration routes; type set; get rid of unused interface;
feature: add and setting tooltips

i0ji committed:
[tag]
